assignment/assignment.hs

```diff
@@ -17,7 +17,7 @@ import Data.Char(intToDigit)
 -- Implement a version vcalled `myreverse` using `foldl`.
 
 -- myreverse :: [a] -> [a]
-
+myreverse a = foldl (\x1 x2 -> x2:x1) [] a
 -------
```

Comment:
Very very nice! Thank you.

One very minor tweak: the function you pass to fold would be the same as `(:)`, but the parameters are the wrong way around, so you could also write

~~~ hs
myreverse = foldl (flip (:)) []
~~~

I also got rid of the parameter on each side, so myreverse returns a function that reverses the list, and not the reversed list itself.